src/main.js

```diff
@@ -154,6 +165,9 @@ async function merge(pull_number) {
     catch (err) {
         core.info("Merge failed.");
         core.debug(err);
+        if (require_merge) {
+            core.setFailed("Merge failed.");
```

Comment:
I think that the fail should **replace** the info, like that:
```js
if (require_merge) {
    core.setFailed("Merge failed");
}
else {
    core.info("Merge failed");
}
```
I think that the following would also work, your call:
```js
core[require_merge ? "setFailed" : "info"]("Merge failed");
```
Also, make sure to print the debug part as the last thing.